src/select-random-repr/select.ml

```diff
@@ -0,0 +1,59 @@
+let () =
+  let ver, output =
+    try
+      match Sys.argv with
+      | [|_; "-ocaml-version"; v; "-o"; fn|] ->
+        (Scanf.sscanf v "%d.%d" (fun major minor -> (major, minor)),
+         fn)
+      | _ -> raise Exit
+    with _ ->
+      failwith "bad command line arguments"
+  in
+  let oc = open_out output in
+  if ver >= (5, 0) then
+    Printf.fprintf oc {|
+module Repr = struct
+  open Caml.Bigarray
+
+  type t = (int64, int64_elt, c_layout) Array1.t
+
+  let of_state : Caml.Random.State.t -> t = Caml.Obj.magic
+end
+
+let assign dst src =
+  let dst = Repr.of_state (Lazy.force dst) in
+  let src = Repr.of_state (Lazy.force src) in
+  Caml.Bigarray.Array1.blit src dst
+
+let make_default default =
+  let split_from_parent v =
+    Lazy.from_val (Caml.Random.State.split (Lazy.force v))
```

Comment:
I've switched it to `Lazy.map_val` - that has I think the exact desired semantics here. If a program has used the PRNG, then domain spawn will _immediately_ split it, otherwise each domain will split on first use of the PRNG in that domain. I'm not at all expert on this, but I don't expect that the statistical difference between splitting the PRNG at domain spawn and splitting at first use of the PRNG in the domain is particularly important, @kayceesrk?

Reply:
Up to you - I'm pretty confident this is correct. If it isn't, then it would just require using `Domain.before_first_spawn` to register a function which ensures the random state is initialised when just before the program spawns its first domain (so single-domain programs could not use randomness and skip initialising the state).